qiskit_experiments/test/t2hahn_backend.py

```diff
@@ -0,0 +1,317 @@
+        self._rng = np.random.default_rng(seed=SEED)
+        self._measurement_error = 0.05
+        super().__init__(configuration)
+
+    @classmethod
+    def _default_options(cls):
+        """Default options of the test backend."""
+        return Options(shots=1024)
+
+    def _qubit_initialization(self) -> dict:
+        if self._initialization_error is None:
+            return {"qubit state": 0, "XY plain": False, "YZ plain": False, "Theta": 0}
+        else:
+            return {
+                "qubit state": (self._rng.random() < self._initialization_error[0]),
+                "XY plain": False,
+                "YZ plain": False,
+                "Theta": 0,
+            }
+
+    def _delay_gate(self, qubit_state: dict, delay: float, t2hahn: float) -> dict:
+        if qubit_state["XY plain"]:
+            prob_noise = 1 - (np.exp(-delay / t2hahn))
+            if self._rng.random() < prob_noise:
+                if self._rng.random() < 0.5:
+                    new_qubit_state = {
+                        "qubit state": 0,
+                        "XY plain": False,
+                        "YZ plain": False,
+                        "Theta": 0,
+                    }
+                else:
+                    new_qubit_state = {
+                        "qubit state": 1,
+                        "XY plain": False,
+                        "YZ plain": False,
+                        "Theta": 0,
+                    }
+            else:
+                phase = self._frequency[0] * delay
+                new_qubit_state = {
+                    "qubit state": qubit_state["qubit state"],
+                    "XY plain": True,
+                    "YZ plain": False,
+                    "Theta": qubit_state["Theta"] + phase,
+                }
+        else:
+            new_qubit_state = qubit_state
+        return new_qubit_state
+
+    def _rx_gate(self, qubit_state: dict, angle: float) -> dict:
+        if qubit_state["XY plain"]:
+            if isclose(angle, np.pi):
+                new_qubit_state = {
+                    "qubit state": 0,
+                    "XY plain": True,
+                    "YZ plain": False,
+                    "Theta": np.pi - qubit_state["Theta"],
+                }
+        elif isclose(qubit_state["qubit state"], 0):
+            if isclose(angle, np.pi):
+                new_qubit_state = {
+                    "qubit state": 1,
+                    "XY plain": False,
+                    "YZ plain": False,
+                    "Theta": 0,
+                }
+        else:
+            if isclose(angle, np.pi):
+                new_qubit_state = {
+                    "qubit state": 0,
+                    "XY plain": False,
+                    "YZ plain": False,
+                    "Theta": 0,
+                }
+        return new_qubit_state
+
+    def _ry_gate(self, qubit_state: dict, angle: float) -> dict:
+        if qubit_state["XY plain"]:
+            if isclose(angle, np.pi / 2):
```

Comment:
I don't see why you need the `if` of the angle and the `if` of theta

Reply:
`angle` is the rotation that the gate need to execute while 'qubit_state["Theta"]' is for the angle in one of the plains (XY, YZ).
I don't need it but, if we want in the future to implement even number of echoes we will need to be sensitive if the `angle `is pi/2 or -pi/2 so I wanted to lay the groundwork for it.
In addition, the `if isclose(qubit_state["Theta"], 0):` is used to determine where are we on the Bloch Sphere.